햄버거 메뉴 수정 및 4개 타입의 아이템 컴포넌트 개발

hjthebunny opened the pull request:
[햄버거 메뉴 - Nav.js]
- css에서 position: absolute과 z-index를 적용해 드롭다운 나타날 시 다른 요소들이 함께 움직이지 않도록 위치를 고정시켰습니다.
- Nav 컴포넌트를 Header 컴포넌트 안에서 호출하여 메뉴 버튼 클릭 시 드롭다운이 나타날 수 있도록 한 컴포넌트에서만 상태 변경이 일어나도록 했습니다. 이를 통해 상태 끌어올리기와 같은 복잡성을 감소시켰습니다. 

[아이템 컴포넌트 - Product.js]
- 하나의 컴포넌트를 개발하되 삼항연산자를 활용하여 각 타입별로 다른 결과물이 나타날 수 있도록 작성했습니다.
- 사진 아래의 텍스트 영역에서 오른쪽 위의 텍스트 색이 타입 별로 달라지는 것은 props를 전달해서 색을 지정해주었습니다.

[기타]
- Main.js의 상품 리스트 영역은 Product 컴포넌트의 출력이 제대로 되고 있는지 확인하기 위해 임시 작성한 것으로, 메인 페이지 구현 시 수정될 예정입니다.

[코드 리뷰 반영한 코드 수정]
- styled-components가 길어질 수록 파일이 복잡해지는 것을 방지하기 위해 컴포넌트 파일과 스타일을 담는 파일을 분리하였습니다.
- 시맨틱한 태그 작성을 위해 태그를 일부 수정했습니다.
- variable.css 파일을 생성해 자주 쓰이는 CSS를 전역 변수로 선언했습니다.
- 열거형 enum 사용하여 아이템 type을 정형화했습니다.

hjthebunny committed:
드롭다운 수정. 북마크 제외 상품 컴포넌트 코드 작성 완료.

api 직접 fetch 하기 전 미리 데이터를 불러와 더미데이터 생성 후 코드 작성에 활용

hjthebunny committed:
코드 정렬, 북마크 아이콘 포함 상품 컴포넌트 완성

hjthebunny committed:
styled-components 들여쓰기 수정, 요소 이름 수정

apeachicetea commented on src/components/Footer.js:
```diff
@@ -2,30 +2,27 @@ import styled from "styled-components";
-    )
+  return (
+    <FooterContainer>
+      <div>
```
div태그로 남기기보다는 styled-components를 이용해서 Text라는 이름 등으로 시멘틱하게 변경해주는 것은 어떨까요?

apeachicetea commented on src/components/Nav.js:
```diff
@@ -4,54 +4,54 @@ import styled from "styled-components";
+right: 0%;
+top: 12.79%;
+bottom: 0%;
+background: #FFFFFF;
```
자주 쓰이는 색상아니 글자의 크기 등의 경우 전역 변수로 선언해서 사용하는 것이 추후 유지 보수시에도 용이합니다.

전역 변수의 경우 아래와 같이 작성하면 됩니다.
`:root {
  --primary-color: #2294e3;
  --border-color: #9a9a9a;
}`

apeachicetea commented on src/components/Product.js:
```diff
@@ -68,7 +76,8 @@ left:0`;
     return (
         <ItemContainer >
-            <ItemImg src={product.type === "Brand" ? product.brand_image_url : product.image_url} alt={product.type === "Brand" ? product.brand_name : product.title} />
+            <ItemImg><img src={product.type === "Brand" ? product.brand_image_url : product.image_url} alt={product.type === "Brand" ? product.brand_name : product.title} />
```
상품 타입에 따른 4가지 분류를 단순히 string값이 아님 enum이나 다른 정형화 된 값으로 관리하는게 더 좋아보입니다.
javascript 에서 객체를 enum처럼 사용할 수 있는 방법에 대해서 검색하여 적용해보세요.

apeachicetea commented on src/components/Product.js:
```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import styled from 'styled-components';
+
+export const ItemContainer = styled.li`
```
이처럼 스타일을 담당하는 코드가 길어질 시 분리하여 관리하게 됩니다.
`components/Product`라는 폴더에 Product.js와 스타일을 담는 파일을 같이 관리하면 추후 유지보수에도 좋습니다.

hjthebunny committed:
코드 리뷰 반영하여 코드 수정

1. [Footer.js] 시맨틱한 태그 작성
2. variables.css 생성하여 자주 쓰이는 CSS 전역 변수 선언
3. 열거형 enum 사용해 상품 타입을 정형화된 값으로 관리
4. 리액트 영역의 코드와 스타일을 담는 코드 파일 분리
